### PR TITLE
Update output to make the root object an overview with results contained inside

### PR DESCRIPTION
**Pull request by gregsdennis**
This makes progress toward some of what is discussed in https://github.com/json-schema-org/json-schema-spec/issues/1065.

Basically, we need a place to include metadata about an evaluation.  This PR wraps the existing output in a root node that can contains any such metadata.

(As a side effect, it creates a purpose for the root object in the case of the "list" format, which could have just been an array before.)

### CODE REVIEW

**Comment by gregsdennis** [comment]
Re-reading through the issue, it seems that `dialect` should be part of the output unit itself rather than the summary.  Still... I think the summary root node is a good addition.

**Review by jdesrosiers — COMMENTED** [review]
> Re-reading through the issue, it seems that `dialect` should be part of the output unit itself rather than the summary. Still... I think the summary root node is a good addition.

I agree that `dialect` should be part of the output unit. I'm not sure putting it in the summary makes sense. Validation can include references to other schema resources that use a different dialect. So, `dialect` in the summary is only telling you the dialect of the root schema. Maybe that's ok, but it might be misleading if the evaluation includes multiple dialects.

**Comment by gregsdennis** [comment]
I'll remove dialect from this PR and do another one for that then.

I know we had a discussion somewhere on having some more "global" details in the root node, though...

File: output/schema.json [hunk 1]
```diff
@@ -3,11 +3,25 @@
...
-  "anyOf": [
-    { "$ref": "#/$defs/flag" },
-    { "$ref": "#/$defs/list" },
-    { "$ref": "#/$defs/hierarchical" }
-  ],
+  "properties": {
+    "dialect": {
+      "type": "string",
+      "format": "iri"
+    },
```
Comment (jdesrosiers) [inline output/schema.json:10]:
```suggestion
```

File: jsonschema-validation-output-machines.md [hunk 2]
```diff
@@ -219,13 +216,14 @@ The passing instance will produce the following annotations:
...
-In the simplest case, merely the boolean result for the `valid` valid property
-needs to be fulfilled. For this format, all other information is explicitly
-omitted.
+In the simplest case, only the overview needs to be fulfilled. For the "flag"
+format, all other information is explicitly omitted.
 
 ```json "Flag Results"
 {
-  "valid": false
+  "dialect": "https://json-schema.org/draft/next/schema",
```
Comment (jdesrosiers) [inline jsonschema-validation-output-machines.md:224]:
```suggestion
```

File: output/schema.json [hunk 3]
```diff
@@ -3,11 +3,25 @@
...
+    "valid": { "type": "boolean" },
+    "details": {
+      "anyOf": [
+        { "$ref": "#/$defs/list" },
+        { "$ref": "#/$defs/hierarchical" }
+      ]
+    }
+  },
+  "requires": [ "dialect", "baseUri", "valid" ],
```
Comment (jdesrosiers) [inline output/schema.json:23]:
```suggestion
  "requires": [ "baseUri", "valid" ],
```

**Review by jdesrosiers — APPROVED** [review]
You missed a couple places removing `dialect`, but otherwise, this looks good to me.

**Comment by gregsdennis** [comment]
Superceded by #1542

**Pull request closed without merging**